crypto/serializer/serializer_local.h

```diff
@@ -50,3 +72,58 @@ struct ossl_serializer_ctx_st {
      */
     UI_METHOD *allocated_ui_method;
 };
+
+struct ossl_deserializer_instance_st {
+    OSSL_DESERIALIZER *deser;    /* Never NULL */
+    void *deserctx;              /* Never NULL */
+    const char *input_type;      /* Never NULL */
+    const char *output_type;     /* Never NULL */
+};
+
+DEFINE_STACK_OF(OSSL_DESERIALIZER_INSTANCE)
+
+struct ossl_deserializer_ctx_st {
+    /*
+     * The caller may know the input type of the data they pass.  If not,
+     * this will remain NULL and the deserializing functionality will try
+     * whatever it has in store.
+     */
+    const char *start_input_type;
+
+    /*
+     * Deserializers that are components of any current deserialization path.
+     */
+    STACK_OF(OSSL_DESERIALIZER_INSTANCE) *deser_insts;
+
+    /*
+     * The finalizer of a deserialization, and its caller argument.
+     */
+    OSSL_DESERIALIZER_FINALIZER *finalizer;
```

Comment:
Well these are a mouthful - wouldnt it be better to keep it consistent with other methods?
i.e.
OSSL_DESERIALIZER_FINAL *final;
OSSL_DESERIALIZER_CLEANUP *cleanup;

Reply:
It's not related the way you seem to assume. But sure.

Reply:
Actually, I'm backing off from that change for now.  If you read the manual on the subject, the text is quite fluent when written in terms of a *finalizer*...  I'm not quite sure how to rewrite that using *final* with the same kind of flow.

Perhaps we need another term entirely instead of *finalize*?  Like I said, this is nowhere near the meaning of "final" as seen in EVP.

Reply:
Constructor? As in object construction?

Reply:
...  "constructor" is less a mouthful than "finalizer"???  :wink:

But, other than that, it is a viable option.  But boy, those function names are getting long...

Reply:
I meant it as being more descriptive not less mouthful.